app/parse.py

```diff
@@ -15,8 +48,110 @@ class Product:
     num_of_reviews: int
 
 
+def product_parse(
+        driver: webdriver.Chrome,
+        product_soup: BeautifulSoup
+) -> Product:
+    detailed_url = urljoin(
+        BASE_URL,
+        product_soup.select_one(".title")["href"]
+    )
+    driver.get(detailed_url)
+    title = (
+        driver.find_element(By.CLASS_NAME, "title")
+        .text.strip()
+    )
+    return Product(
+        title=title,
+        description=product_soup.find(
+            "p", {"class": "description"}
+        )
+        .text.strip()
+        .replace("\xa0", " ")
+        .strip(),
+        price=float(
+            product_soup.find("h4", {"class": "price"})
+            .text.strip()
+            .replace("$", "")
+        ),
+        rating=int(len(product_soup.find_all(
+            "span", class_="ws-icon-star"))
+        ),
+        num_of_reviews=int(
+            product_soup.find("p", {"class": "review-count"})
+            .text.strip()
+            .split()[0]
+        ),
+    )
+
+
+def load_full_page(
+        driver: webdriver.Chrome,
+        url: str
+) -> BeautifulSoup:
+    driver.get(url)
+
+    try:
+        accept_button = WebDriverWait(driver, 2).until(
+            exc.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+        )
+        accept_button.click()
+        print("Cookies accepted")
+    except (TimeoutException, NoSuchElementException):
+        print("Cookies not found")
+
+    while True:
+        try:
+            more_button = WebDriverWait(driver, 2).until(
+                exc.presence_of_element_located(
+                    (By.CLASS_NAME, "ecomerce-items-scroll-more")
+                )
+            )
```

Comment:
Similar to the previous comment, the timeout for waiting for the 'more_button' is also set to 2 seconds. Ensure this is sufficient for the website's response time to avoid unnecessary exceptions.